### PR TITLE
FIPS: Remove ability to bypass the FIPS self tests

### PR DESCRIPTION
**Pull request by slontis**
This is a FIPS 140-3 requirement.
It should not be done as a FIPS indicator.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
fipsinstall related documentation is in #23689

**Comment by paulidale** [comment]
Why do we need to retain the `#ifdef`ed code?

File: providers/fips/fipsprov.c [hunk 1]
```diff
@@ -214,32 +214,21 @@ static int fips_get_params_from_core(FIPS_GLOBAL *fgbl)
...
-            sizeof(fgbl->selftest_params.indicator_data));
-    *p++ = OSSL_PARAM_construct_utf8_ptr(
-            OSSL_PROV_FIPS_PARAM_INSTALL_VERSION,
-            (char **)&fgbl->selftest_params.indicator_version,
-            sizeof(fgbl->selftest_params.indicator_version));
-    *p++ = OSSL_PARAM_construct_utf8_ptr(
-            OSSL_PROV_FIPS_PARAM_CONDITIONAL_ERRORS,
-            (char **)&fgbl->selftest_params.conditional_error_check,
-            sizeof(fgbl->selftest_params.conditional_error_check));
+    OSSL_PARAM core_params[27], *p = core_params;
```
Comment (paulidale) [inline providers/fips/fipsprov.c:217]: This will conflict in due course :(

Comment (slontis) [reply providers/fips/fipsprov.c:217]: yep...

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Rebased with no changes

File: providers/fips/fipsprov.c [hunk 2]
```diff
@@ -218,7 +218,7 @@ static int fips_get_params_from_core(FIPS_GLOBAL *fgbl)
     * OSSL_PROV_FIPS_PARAM_SECURITY_CHECKS and
     * OSSL_PROV_FIPS_PARAM_TLS1_PRF_EMS_CHECK are not self test parameters.
     */
-    OSSL_PARAM core_params[32], *p = core_params;
+    OSSL_PARAM core_params[28], *p = core_params;
```
Comment (nhorman) [inline providers/fips/fipsprov.c:221]: Its odd, our cross compile targets seem to complain about this being too short an array, but not our non-cross builds.  That doesn't make sense

Comment (paulidale) [reply providers/fips/fipsprov.c:221]: I too wondered why the compiler checks were removed in more recent versions.
My interest is not sufficiently piqued to dive into the gcc change log.

Comment (slontis) [reply providers/fips/fipsprov.c]: obviously my merge didnt go exactly as planned :)

**Comment by slontis** [comment]
PR https://github.com/openssl/openssl/pull/25159 fixes the test issue

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
It needs a rebase after the ecdh cofactor indicator PR went in.

**Comment by slontis** [comment]
Merged: thanks for reviewing.

**Pull request closed without merging**